### ISSUE TITLE
False positive for disk caching when no VM data disks is null in templates

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#338]
**Description of the issue**

Within a template, if a virtual machine has no data disks and the `properties.storageprofile.dataDisks` property does not exist at all, a false positive is generated by `Azure.VM.DiskCaching`.

The `dataDisks` property may not exist in a template.

**Expected behaviour**

Update rule conditions to handle `dataDisks` being null.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.10.0**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
